### DISCUSSION TITLE
Kubernetes feasibility

### DISCUSSION THREAD

**Discussion by coopbri** [prisma/prisma#6562] [Q&A] [answered]
Hello,

I am currently running Prisma with Postgres in a monolith (stateless), and am decomposing it into microservices within Kubernetes. I successfully deployed the monolith within a pod, with Postgres external to the cluster.

My question is: is it feasible to run Prisma, in its current state, within a distributed architecture? I am curious about DB relations/handling joins, splitting up the Prisma schema, connections/pooling, and things like that. I would love to hear people's approaches to running Prisma in Docker/Kubernetes.

**UPDATE:** I "solved" this by creating a small shell script that concatenates multiple Prisma schema files containing service-specific models. Of course, this approach has the drawback of high coupling between microservices, so it is not ideal nor as modular as it could be. A similar (probably better) tool to accomplish this would be [prisma-merge](https://www.npmjs.com/package/prisma-merge). I would still love to hear approaches from others.

To see progress towards native support for multiple Prisma schema files, track #2377. I also recommend keeping an eye on Prisma CockroachDB driver support (#2523).

**Comment by ryands17** [comment, marked as answer]
@coopbri 👋 

Yes that workaround will be the only thing that can be done for now. Another thing is that you can develop your Prisma schema as an NPM package so that it is the single source of truth and then import that in each micro service so that there is consistency b/w the services.

As for the CockroachDB driver, yes it would be great to have that as it is best suited for a micro service architecture. I would also suggest following this issue for more info as it matches your use-case https://github.com/prisma/prisma/issues/5000.

**Reply by coopbri** [reply]
Thanks for your response, @ryands17!

I appreciate the confirmation about the process for maintaining the schema as a single source of truth. Thank you for sharing that issue about inter-service communication, I will keep an eye on that.

And one more thing: Prisma is excellent, I use it in production in several applications. Thank you for your hard work on it -- I love how friendly and responsive the Prisma community is.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
